numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -621,6 +594,31 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_str(PyObject *self){
+    PyObject *arr, *ret = NULL;
+
+    if (!PyDataType_HASFIELDS(((PyVoidScalarObject*)self)->descr)) {
+        PyObject  *item, *item_str;
+
+        item = gentype_generic_method(self, NULL, NULL, "item");
+        if (item == NULL) {
+            return NULL;
+        }
+
+        item_str = PyObject_Str(item);
+        Py_DECREF(item);
+        return item_str;
+    }
```

Comment:
It does? I don't see that:
```python
>>> np.zeros(2, dtype='f,f')
array([(0., 0.), (0., 0.)],
      dtype=[('f0', '<f4'), ('f1', '<f4')])
```

Also, this code is inside the non-fields clause so should only affect raw void items.